### ISSUE TITLE
Transaction decorator does not match decorated function return value

### ISSUE THREAD

**Issue by tjsousa** [appsignal/appsignal-elixir#164]
When using the decorated strategy within handler functions for `GenServer/GenEvent`, the decorated function return value is not propagated to OTP, leading to a crash.

Here's the workaround:

```elixir
      def handle_event(%{"event" => event_name} = event, state) do
        transaction = Appsignal.Transaction.start(
          Appsignal.Transaction.generate_id,
          :background_job
        )
        |> Appsignal.Transaction.set_action("#{__MODULE__}##{event_name}")

        result = process(event, state)

        Appsignal.Transaction.finish(transaction)
        :ok = Appsignal.Transaction.complete(transaction)

        result
      end
```

**Comment by arjan** [comment]
Do you have an example how you use the decorator?

**Comment by tjsousa** [comment]
Hi @arjan.

I used it like this:

```elixir
defmodule EventHandler do
      use GenEvent
      use Appsignal.Instrumentation.Decorators

      @decorate transaction(:background_job)
      def handle_event(%{"event" => event_name} = event, state) do
         process(event, state)
      end

      def process(%{"event" => "event_type", state} do
        # some processing for this type of event
        {:ok, state}
      end
end
```

The error I get is directly from `GenEvent`:
```
220 <190>1 2017-03-02T18:00:50.236349+00:00 app consumer.1 - - 2017-03-02T18:00:50.236Z [ level="error"] GenEvent handler AuthConsumer.LogHandler installed in #PID<0.479.0> terminating
130 <190>1 2017-03-02T18:00:50.236351+00:00 app consumer.1 - - ** (stop) bad return value: :ok
```

**Comment by arjan** [comment]
I've committed the fix. @jeffkreeftmeijer when are you guys doing the next release? :)

**Comment by jeffkreeftmeijer** [comment]
Thanks @arjan! I've just released [1.0.4](https://hex.pm/packages/appsignal/1.0.4) to get this patch out there. @tjsousa could you try 1.0.4 and let us know if it solved your issue? :)

**Comment by tjsousa** [comment]
@jeffkreeftmeijer I've tested it in 1.0.4 and it works as expected.

Thank you very much @arjan and kudos for the quick release. You guys rock!
